Add config properties for HTTP security headers

jportner opened the pull request:
Resolves #52809 and #97153.

### Overview

This PR adds the following config properties (each prefixed by `server.securityResponseHeaders`):

| Config property | Default value | Notes |
|-|-|-|
| `.strictTransportSecurity` | `null` (Header is _not_ enabled) | This controls the [Strict-Transport-Security](https://scotthelme.co.uk/hsts-the-missing-link-in-tls/) header, which is an excellent feature to support on your site and strengthens your implementation of TLS by getting the User Agent to enforce the use of HTTPS. <br />Can be set to `null` to disable this header. |
| `.xContentTypeOptions` | `'nosniff'` (Header is enabled) | This controls the [X-Content-Type-Options](https://scotthelme.co.uk/hardening-your-http-response-headers/#x-content-type-options) header, which stops a browser from trying to MIME-sniff the content type and forces it to stick with the declared content-type. The only valid value for this header is "X-Content-Type-Options: nosniff". <br />Can be set to `null` to disable this header. |
| `.referrerPolicy` | `'no-referrer-when-downgrade'` (Header is enabled) | This controls the [Referrer-Policy](https://scotthelme.co.uk/a-new-security-header-referrer-policy/) header, which is a new header that allows a site to control how much information the browser includes with navigations away from a document and should be set by all sites. <br />Can be set to `null` to disable this header. |
| `.permissionsPolicy` | `null` (Header is _not_ enabled) | This controls the [Permissions-Policy](https://scotthelme.co.uk/goodbye-feature-policy-and-hello-permissions-policy/) header, which is a new header that allows a site to control which features and APIs can be used in the browser. <br />Can be set to `null` to disable this header. |
| `.disableEmbedding` | `false` (Header is _not_ enabled, embedding is allowed) | This influences the default [Content-Security-Policy](https://scotthelme.co.uk/content-security-policy-an-introduction/) header and controls the [X-Frame-Options](https://scotthelme.co.uk/hardening-your-http-response-headers/#x-frame-options) header, which tells the browser whether you want to allow your site to be framed or not. By preventing a browser from framing your site you can defend against attacks like clickjacking. |

#### New scan result from securityheaders.com with default settings ("C" rating):

![image](https://user-images.githubusercontent.com/5295965/114961225-9cbd3400-9e36-11eb-81ee-64c48f06c6f1.png)

### Docs

**Docs preview**: [link](https://kibana_97158.docs-preview.app.elstc.co/guide/en/kibana/master/settings.html#server-securityResponseHeaders-strictTransportSecurity) ([diff](https://kibana_97158.docs-preview.app.elstc.co/diff))

**Note concerning the docs**: the asciidoc hacks that I needed to make for the settings table in bb8e2aa are kind of awful, but it seems to be the only solution that will work. The key names for the new config options are too long, and they stretch the left column of the table out to the maximum size as a result. Regular columns in AsciiDoc will not preserve leading whitespace characters, so I had to utilize an [AsciiDoc block element column style](https://docs.asciidoctor.org/asciidoc/latest/tables/format-column-content/) with the `a` operator.

#### Screenshots before and after bb8e2aa

Before:

![image](https://user-images.githubusercontent.com/5295965/114811653-7cc93a00-9d7c-11eb-983c-8d4efbbf7a12.png)

After: 

![image](https://user-images.githubusercontent.com/5295965/114811841-d7629600-9d7c-11eb-8640-c9df746abf11.png)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119838/)
* Commit: 62864b136641ba2cc346620bd6c75d8048a67cfb
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Jest Integration Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119838/execution/node/444/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119838/","number":"119838","commit":"62864b136641ba2cc346620bd6c75d8048a67cfb"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119858/)
* Commit: bb8e2aafcd0d7e73742d6956521c9f693a4fc596
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-97158/bb8e2aafcd0d7e73742d6956521c9f693a4fc596)
* [Documentation Changes](https://kibana_97158.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #119838](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119838/) failed 62864b136641ba2cc346620bd6c75d8048a67cfb

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119838/","number":"119838","commit":"62864b136641ba2cc346620bd6c75d8048a67cfb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119858/","number":"119858","commit":"bb8e2aafcd0d7e73742d6956521c9f693a4fc596"}]}
PIPELINE-->

jportner commented on src/core/server/http/lifecycle_handlers.ts:
```diff
@@ -62,12 +62,12 @@ export const createVersionCheckPostAuthHandler = (kibanaVersion: string): OnPost
 
   return (request, response, toolkit) => {
     const additionalHeaders = {
-      ...customHeaders,
+      ...securityResponseHeaders,
+      ...customResponseHeaders,
       [KIBANA_NAME_HEADER]: serverName,
     };
```
We explicitly allow `customResponseHeaders` to override any values in `securityResponseHeaders`. This is so the new config properties do not break existing configurations that already set the same headers using `customResponseHeaders`.

We could detect this situation when the config is set and throw an error, but we'd have to make that change in the 8..0 release, I think.

jportner commented on src/core/server/http/security_response_headers_config.ts:
```diff
@@ -0,0 +1,84 @@
+
+import { schema, TypeOf } from '@kbn/config-schema';
+
+const strictTransportSecuritySchema = schema.object({
+  // See: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security
+  maxAge: schema.duration({ defaultValue: '1Y' }),
+  includeSubDomains: schema.boolean({ defaultValue: false }),
+  preload: schema.boolean({ defaultValue: false }),
+});
```
By default, `Strict-Transport-Security` is enabled with a max age of 1 year. This is recommended by MDN and most other authoritative sources. We _could_ decrease this but I'm not sure we should.

I omitted `includeSubDomains` by default because of https://github.com/elastic/kibana/issues/52809#issuecomment-564764345.

Note: if Kibana is served over HTTP, this header is still sent, but browsers ignore it. It only affects browsers once they see this header in an HTTPS response.

jportner commented on src/core/server/http/security_response_headers_config.ts:
```diff
@@ -0,0 +1,84 @@
+  strictTransportSecurity: schema.oneOf([strictTransportSecuritySchema, schema.literal(null)], {
+    defaultValue: strictTransportSecuritySchema.validate({}),
+  }),
+  xContentTypeOptions: schema.oneOf([schema.literal('nosniff'), schema.literal(null)], {
+    // See: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Content-Type-Options
+    defaultValue: 'nosniff',
+  }),
```
By default, `X-Content-Type-Options` is enabled. It only has one valid value. This shouldn't have any negative impact on users.

jportner commented on src/core/server/http/security_response_headers_config.ts:
```diff
@@ -0,0 +1,84 @@
+    // See: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Content-Type-Options
+    defaultValue: 'nosniff',
+  }),
+  referrerPolicy: schema.oneOf(
+    // See: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Referrer-Policy
+    [
+      schema.literal('no-referrer'),
+      schema.literal('no-referrer-when-downgrade'),
+      schema.literal('origin'),
+      schema.literal('origin-when-cross-origin'),
+      schema.literal('same-origin'),
+      schema.literal('strict-origin'),
+      schema.literal('strict-origin-when-cross-origin'),
+      schema.literal('unsafe-url'),
+      schema.literal(null),
+    ],
+    { defaultValue: 'no-referrer-when-downgrade' }
+  ),
```
By default, `Referrer-Policy` is enabled with a value of "no-referrer-when-downgrade". This is not very strict, it does the bare minimum to prevent the Referer header from being sent to unsafe (HTTP) destinations when Kibana is served over HTTPS. See also https://github.com/elastic/kibana/issues/52809#issuecomment-564788028

jportner commented on src/core/server/http/security_response_headers_config.ts:
```diff
@@ -0,0 +1,84 @@
+    ],
+    { defaultValue: 'no-referrer-when-downgrade' }
+  ),
+  permissionsPolicy: schema.oneOf([schema.string(), schema.literal(null)], {
+    defaultValue: 'camera=(), microphone=()', // disables camera and microphone access in Kibana and any embedded pages
+  }),
```
By default, `Permissions-Policy` is enabled with a value of "camera=(), microphone=()". There is not a lot of authoritative info on this header yet ([the header was previously `Feature-Policy` but has been slightly reworked and renamed](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Feature-Policy)). I figured we can set it with some safe feature policies as a starting point.

Note that `Permissions-Policy` is _subtractive_, so if we do not specify a directive, that is not affected. Kibana shouldn't ever need access to a user's camera or microphone, so what we are saying here is that Kibana (and any embedded pages) are **not allowed** to access these features.

See also: https://github.com/w3c/webappsec-permissions-policy/blob/main/permissions-policy-explainer.md

kobelb replied:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Feature-Policy#example leads me to believe that we should be using `Permissions-Policy: camera 'none'; microphone 'none'`. It also looks like this isn't very widely supported per https://caniuse.com/permissions-policy. Perhaps we should just hold off on this one until it matures? Does the lack of this header get flagged by scanners right now?

jportner replied:
The page is a bit misleading / outdated. That's the format for the old `Feature-Policy` header, it hasn't been updated to the `Permissions-Policy` format.

At any rate based on https://github.com/elastic/kibana/pull/97158#discussion_r614306674 below, I think we will leave this header disabled by default and mark the config prop as experimental, if that sounds good to you.

kobelb replied:
Sounds good to me! Thanks.

jportner commented on src/core/server/http/security_response_headers_config.ts:
```diff
@@ -0,0 +1,84 @@
+  permissionsPolicy: schema.oneOf([schema.string(), schema.literal(null)], {
+    defaultValue: 'camera=(), microphone=()', // disables camera and microphone access in Kibana and any embedded pages
+  }),
+  disableEmbedding: schema.boolean({ defaultValue: false }), // is used to control X-Frame-Options and CSP headers
```
When enabled, this changes the default `Content-Security-Policy` header (by adding the `frame-ancestors` directive) and adds the `X-Frame-Options` header.

Even though it appears that [all of Kibana's target browsers support `frame-ancestors`](https://caniuse.com/mdn-http_headers_csp_content-security-policy_frame-ancestors), we still include `X-Frame-Options` for a couple of reasons:
1. If the user has customized the `Content-Security-Policy` for some reason, we won't add the `frame-ancestors` directive -- so including `X-Frame-Options` as a backup is a good idea
2. Naive security scanning tools can see a missing `X-Frame-Options` as a problem, even if there is a `Content-Security-Policy` present with a `frame-ancestors` directive

In the future (8.0 release?) I would love to change this to be enabled by default.

jportner commented on src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker:
```diff
@@ -116,12 +116,20 @@ kibana_vars=(
     server.compression.referrerWhitelist
     server.cors
     server.cors.origin
-    server.customResponseHeaders
```
This was a duplicate value so I removed it (it still exists below on line 127)

jportner commented on src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker:
```diff
@@ -116,12 +116,20 @@ kibana_vars=(
     server.defaultRoute
     server.host
     server.keepAliveTimeout
     server.maxPayloadBytes
+    server.maxPayload
```
I added `server.maxPayload` but left the deprecated `server.maxPayloadBytes` for now (see 8cc60bf)

jportner commented on src/core/server/csp/csp_config.ts:
```diff
@@ -58,9 +68,13 @@ export class CspConfig implements ICspConfig {
     this.warnLegacyBrowsers = source.warnLegacyBrowsers;
-    this.header = source.rules.join('; ');
+    this.disableEmbedding = source.disableEmbedding;
+    if (!rawCspConfig.rules?.length && source.disableEmbedding) {
+      this.rules.push(FRAME_ANCESTORS_RULE);
+    }
```
It might be a good idea to log a warning if `disableEmbedding` is enabled but custom CSP rules are used that do not include the `frame-ancestors` directive. But I think we can leave that for a future enhancement, as we don't have the ability to log when config is ingested at the moment.

jportner reviewed (commented):
Author's notes for reviewers

jbudz reviewed (approved):
kibana-docker LGTM!

KOTungseth commented on docs/setup/settings.asciidoc:
```diff
@@ -504,6 +504,87 @@ deprecation warning at startup. This setting cannot end in a slash (`/`).
+  permissionsPolicy:
+----
+| Controls whether the https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Feature-Policy[`Permissions-Policy`] header is used in all
+responses to the client from the {kib} server, and specifies what value is used. Allowed values are any text value or `null`. Set this to
```
```suggestion
responses to the client from the {kib} server, and specifies what value is used. Allowed values are any text value or `null`.
```

KOTungseth commented on docs/setup/settings.asciidoc:
```diff
@@ -504,6 +504,87 @@ deprecation warning at startup. This setting cannot end in a slash (`/`).
+  disableEmbedding:
+----
+| Controls whether the https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy[`Content-Security-Policy`] and
+https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options[`X-Frame-Options`] headers will be configured to disable embedding
```
```suggestion
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options[`X-Frame-Options`] headers are configured to disable embedding
```

KOTungseth commented on docs/setup/settings.asciidoc:
```diff
@@ -504,6 +504,87 @@ deprecation warning at startup. This setting cannot end in a slash (`/`).
+----
+| Controls whether the https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy[`Content-Security-Policy`] and
+https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options[`X-Frame-Options`] headers will be configured to disable embedding
+{kib} in other webpages using iframes. Set this to `true` to use secure headers that disable embedding; this will add the
```
```suggestion
{kib} in other webpages using iframes. When set to `true`, secure headers are used to disable embedding, which adds the
```

thomheymann commented on docs/setup/settings.asciidoc:
```diff
@@ -504,6 +504,87 @@ deprecation warning at startup. This setting cannot end in a slash (`/`).
+----
+| Specifies the `max-age` directive of the
+https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security[`Strict-Transport-Security`] response header. This is
+rounded down to the nearest second. *Default:* `'1Y'` (31536000 seconds)
```
As discussed, the settings here are a bit inconsistent right now with the transport policy header being controlled using individual settings whereas permissions policy having to be set manually as the raw header value.

Would be good to make that consistent. Maybe the easiest thing would be to let users set the raw header value themselves so that we don't have to keep updating it manually when the spec changes.

jportner replied:
Agreed, done in c358373!

KOTungseth commented on docs/setup/settings.asciidoc:
```diff
@@ -504,6 +504,87 @@ deprecation warning at startup. This setting cannot end in a slash (`/`).
+| Controls whether the https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy[`Content-Security-Policy`] and
+https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options[`X-Frame-Options`] headers will be configured to disable embedding
+{kib} in other webpages using iframes. Set this to `true` to use secure headers that disable embedding; this will add the
+`frame-ancestors: 'self'` directive to the `Content-Security-Policy` response header (if you are using the default CSP rules), and it will
```
```suggestion
`frame-ancestors: 'self'` directive to the `Content-Security-Policy` response header (if you are using the default CSP rules), and adds
```

KOTungseth commented on docs/setup/settings.asciidoc:
```diff
@@ -504,6 +504,87 @@ deprecation warning at startup. This setting cannot end in a slash (`/`).
+https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options[`X-Frame-Options`] headers will be configured to disable embedding
+{kib} in other webpages using iframes. Set this to `true` to use secure headers that disable embedding; this will add the
+`frame-ancestors: 'self'` directive to the `Content-Security-Policy` response header (if you are using the default CSP rules), and it will
+add the `X-Frame-Options: SAMEORIGIN` response header as well. *Default:* `false`
```
```suggestion
the `X-Frame-Options: SAMEORIGIN` response header. *Default:* `false`
```

KOTungseth commented on docs/setup/settings.asciidoc:
```diff
@@ -517,6 +598,10 @@ back end server. To allow remote users to connect, set the value to the IP addre
 the <<server-socketTimeout, `server.socketTimeout`>> counter. *Default: `"120000"`*
 
 |[[server-maxPayloadBytes]] `server.maxPayloadBytes:`
+ | deprecated:[7.13.0,"This setting will no longer be supported as of 8.0."]
```
```suggestion
 | deprecated:[7.13.0,"In 8.0 and later, this setting will no longer be supported."]
```

jportner replied:
Thanks! There are a couple of other config properties with the same string (`cpu.cgroup.path.override` and `cpuacct.cgroup.path.override`), I'll update those too.

thomheymann commented on docs/setup/settings.asciidoc:
```diff
@@ -504,6 +504,87 @@ deprecation warning at startup. This setting cannot end in a slash (`/`).
+a|
+----
+server.securityResponseHeaders:
+  strictTransportSecurity.preload:
```
@elastic/kibana-docs  What's the best way of printing long settings like this? The table is too narrow to fit both the name and description. However, breaking it up like this makes it a bit confusing for users too.

KOTungseth replied:
@thomheymann the table capabilities in Asciidoc are limited, and I agree that breaking up the setting names is confusing. These would be best displayed in a definition list instead of a table. I propose that we merge these changes, then I'll fix the formatting before 7.13 releases.

KOTungseth reviewed (approved):
I have a handful of nits, but otherwise, LGTM!

thomheymann reviewed (approved):
Looks great! Tested locally and can confirm that Kibana responds with correct security headers. 

Couple minor comments below.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120056/)
* Commit: 84828e34f41a3fc0a8928733b95810a54dcc21c5
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120056/execution/node/426/log)

### History
* :green_heart: [Build #119858](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119858/) succeeded bb8e2aafcd0d7e73742d6956521c9f693a4fc596
* :broken_heart: [Build #119838](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119838/) failed 62864b136641ba2cc346620bd6c75d8048a67cfb

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119838/","number":"119838","commit":"62864b136641ba2cc346620bd6c75d8048a67cfb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119858/","number":"119858","commit":"bb8e2aafcd0d7e73742d6956521c9f693a4fc596"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120056/","number":"120056","commit":"84828e34f41a3fc0a8928733b95810a54dcc21c5"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120061/)
* Commit: 34d1650a6d0c8bd9c37c6150d0bd8f7441464991
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-97158/34d1650a6d0c8bd9c37c6150d0bd8f7441464991)
* [Documentation Changes](https://kibana_97158.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/6c07eb238212b8d6b48579640e8374cef5046d36) | [after](https://github.com/elastic/kibana/commit/34d1650a6d0c8bd9c37c6150d0bd8f7441464991) | diff |
| --- | --- | --- | --- |
| `actions` | - | 113 | +113 |
| `advancedSettings` | - | 18 | +18 |
| `alerting` | - | 200 | +200 |
| `apm` | - | 35 | +35 |
| `apmOss` | - | 38 | +38 |
| `banners` | - | 9 | +9 |
| `beatsManagement` | - | 2 | +2 |
| `bfetch` | - | 58 | +58 |
| `canvas` | - | 4 | +4 |
| `cases` | - | 2 | +2 |
| `charts` | - | 157 | +157 |
| `cloud` | - | 19 | +19 |
| `console` | - | 2 | +2 |
| `core` | - | 2170 | +2170 |
| `dashboard` | - | 138 | +138 |
| `dashboardEnhanced` | - | 51 | +51 |
| `dashboardMode` | - | 11 | +11 |
| `data` | - | 3436 | +3436 |
| `dataEnhanced` | - | 52 | +52 |
| `devTools` | - | 9 | +9 |
| `discover` | - | 61 | +61 |
| `discoverEnhanced` | - | 39 | +39 |
| `embeddable` | - | 426 | +426 |
| `embeddableEnhanced` | - | 14 | +14 |
| `encryptedSavedObjects` | - | 20 | +20 |
| `enterpriseSearch` | - | 2 | +2 |
| `esUiShared` | - | 84 | +84 |
| `eventLog` | - | 70 | +70 |
| `expressions` | - | 1755 | +1755 |
| `features` | - | 199 | +199 |
| `fileUpload` | - | 175 | +175 |
| `fleet` | - | 1075 | +1075 |
| `globalSearch` | - | 67 | +67 |
| `home` | - | 91 | +91 |
| `indexLifecycleManagement` | - | 5 | +5 |
| `indexManagement` | - | 162 | +162 |
| `indexPatternFieldEditor` | - | 29 | +29 |
| `indexPatternManagement` | - | 46 | +46 |
| `infra` | - | 14 | +14 |
| `ingestPipelines` | - | 9 | +9 |
| `inspector` | - | 97 | +97 |
| `kibanaLegacy` | - | 92 | +92 |
| `kibanaReact` | - | 232 | +232 |
| `kibanaUtils` | - | 465 | +465 |
| `lens` | - | 160 | +160 |
| `licenseManagement` | - | 3 | +3 |
| `licensing` | - | 116 | +116 |
| `lists` | - | 251 | +251 |
| `management` | - | 35 | +35 |
| `maps` | - | 182 | +182 |
| `mapsEms` | - | 69 | +69 |
| `ml` | - | 343 | +343 |
| `monitoring` | - | 9 | +9 |
| `navigation` | - | 29 | +29 |
| `newsfeed` | - | 17 | +17 |
| `observability` | - | 157 | +157 |
| `osquery` | - | 8 | +8 |
| `presentationUtil` | - | 69 | +69 |
| `remoteClusters` | - | 4 | +4 |
| `reporting` | - | 127 | +127 |
| `rollup` | - | 20 | +20 |
| `ruleRegistry` | - | 19 | +19 |
| `runtimeFields` | - | 22 | +22 |
| `savedObjects` | - | 173 | +173 |
| `savedObjectsManagement` | - | 91 | +91 |
| `savedObjectsTagging` | - | 54 | +54 |
| `savedObjectsTaggingOss` | - | 81 | +81 |
| `security` | - | 79 | +79 |
| `securityOss` | - | 11 | +11 |
| `securitySolution` | - | 99 | +99 |
| `share` | - | 66 | +66 |
| `snapshotRestore` | - | 22 | +22 |
| `spaces` | - | 82 | +82 |
| `spacesOss` | - | 64 | +64 |
| `stackAlerts` | - | 4 | +4 |
| `taskManager` | - | 41 | +41 |
| `telemetry` | - | 41 | +41 |
| `telemetryCollectionManager` | - | 24 | +24 |
| `telemetryCollectionXpack` | - | 1 | +1 |
| `telemetryManagementSection` | - | 13 | +13 |
| `timelines` | - | 6 | +6 |
| `triggersActionsUi` | - | 221 | +221 |
| `uiActions` | - | 127 | +127 |
| `uiActionsEnhanced` | - | 185 | +185 |
| `uptime` | - | 5 | +5 |
| `urlForwarding` | - | 14 | +14 |
| `usageCollection` | - | 53 | +53 |
| `visTypeTimeseries` | - | 7 | +7 |
| `visualizations` | - | 212 | +212 |
| total |  |  | +15169 |

#### API count missing comments
| id | [before](https://github.com/elastic/kibana/commit/6c07eb238212b8d6b48579640e8374cef5046d36) | [after](https://github.com/elastic/kibana/commit/34d1650a6d0c8bd9c37c6150d0bd8f7441464991) | diff |
| --- | --- | --- | --- |
| `actions` | - | 113 | +113 |
| `advancedSettings` | - | 17 | +17 |
| `alerting` | - | 200 | +200 |
| `apm` | - | 35 | +35 |
| `apmOss` | - | 38 | +38 |
| `banners` | - | 9 | +9 |
| `beatsManagement` | - | 2 | +2 |
| `bfetch` | - | 47 | +47 |
| `canvas` | - | 3 | +3 |
| `cases` | - | 2 | +2 |
| `charts` | - | 132 | +132 |
| `cloud` | - | 19 | +19 |
| `console` | - | 2 | +2 |
| `core` | - | 990 | +990 |
| `dashboard` | - | 126 | +126 |
| `dashboardEnhanced` | - | 50 | +50 |
| `dashboardMode` | - | 11 | +11 |
| `data` | - | 2944 | +2944 |
| `dataEnhanced` | - | 34 | +34 |
| `devTools` | - | 8 | +8 |
| `discover` | - | 48 | +48 |
| `discoverEnhanced` | - | 37 | +37 |
| `embeddable` | - | 361 | +361 |
| `embeddableEnhanced` | - | 14 | +14 |
| `encryptedSavedObjects` | - | 18 | +18 |
| `enterpriseSearch` | - | 2 | +2 |
| `esUiShared` | - | 82 | +82 |
| `eventLog` | - | 70 | +70 |
| `expressions` | - | 1364 | +1364 |
| `features` | - | 89 | +89 |
| `fileUpload` | - | 175 | +175 |
| `fleet` | - | 985 | +985 |
| `globalSearch` | - | 13 | +13 |
| `home` | - | 67 | +67 |
| `indexLifecycleManagement` | - | 5 | +5 |
| `indexManagement` | - | 157 | +157 |
| `indexPatternFieldEditor` | - | 27 | +27 |
| `indexPatternManagement` | - | 46 | +46 |
| `infra` | - | 11 | +11 |
| `ingestPipelines` | - | 9 | +9 |
| `inspector` | - | 77 | +77 |
| `kibanaLegacy` | - | 84 | +84 |
| `kibanaReact` | - | 207 | +207 |
| `kibanaUtils` | - | 301 | +301 |
| `lens` | - | 149 | +149 |
| `licenseManagement` | - | 3 | +3 |
| `licensing` | - | 40 | +40 |
| `lists` | - | 234 | +234 |
| `management` | - | 35 | +35 |
| `maps` | - | 181 | +181 |
| `mapsEms` | - | 69 | +69 |
| `ml` | - | 339 | +339 |
| `monitoring` | - | 9 | +9 |
| `navigation` | - | 29 | +29 |
| `newsfeed` | - | 17 | +17 |
| `observability` | - | 157 | +157 |
| `osquery` | - | 8 | +8 |
| `presentationUtil` | - | 67 | +67 |
| `remoteClusters` | - | 4 | +4 |
| `reporting` | - | 126 | +126 |
| `rollup` | - | 20 | +20 |
| `ruleRegistry` | - | 19 | +19 |
| `runtimeFields` | - | 17 | +17 |
| `savedObjects` | - | 159 | +159 |
| `savedObjectsManagement` | - | 80 | +80 |
| `savedObjectsTagging` | - | 50 | +50 |
| `savedObjectsTaggingOss` | - | 42 | +42 |
| `security` | - | 38 | +38 |
| `securityOss` | - | 8 | +8 |
| `securitySolution` | - | 90 | +90 |
| `share` | - | 60 | +60 |
| `snapshotRestore` | - | 22 | +22 |
| `spaces` | - | 65 | +65 |
| `spacesOss` | - | 29 | +29 |
| `stackAlerts` | - | 4 | +4 |
| `taskManager` | - | 16 | +16 |
| `telemetry` | - | 37 | +37 |
| `telemetryCollectionManager` | - | 24 | +24 |
| `telemetryCollectionXpack` | - | 1 | +1 |
| `telemetryManagementSection` | - | 12 | +12 |
| `timelines` | - | 6 | +6 |
| `triggersActionsUi` | - | 212 | +212 |
| `uiActions` | - | 88 | +88 |
| `uiActionsEnhanced` | - | 135 | +135 |
| `uptime` | - | 5 | +5 |
| `urlForwarding` | - | 14 | +14 |
| `usageCollection` | - | 46 | +46 |
| `visTypeTimeseries` | - | 7 | +7 |
| `visualizations` | - | 194 | +194 |
| total |  |  | +11998 |

#### API count with any type
| id | [before](https://github.com/elastic/kibana/commit/6c07eb238212b8d6b48579640e8374cef5046d36) | [after](https://github.com/elastic/kibana/commit/34d1650a6d0c8bd9c37c6150d0bd8f7441464991) | diff |
| --- | --- | --- | --- |
| `bfetch` | - | 1 | +1 |
| `charts` | - | 2 | +2 |
| `core` | - | 147 | +147 |
| `dashboard` | - | 1 | +1 |
| `data` | - | 81 | +81 |
| `embeddable` | - | 2 | +2 |
| `esUiShared` | - | 4 | +4 |
| `expressions` | - | 44 | +44 |
| `fileUpload` | - | 6 | +6 |
| `fleet` | - | 16 | +16 |
| `indexManagement` | - | 12 | +12 |
| `inspector` | - | 6 | +6 |
| `kibanaLegacy` | - | 3 | +3 |
| `kibanaUtils` | - | 3 | +3 |
| `maps` | - | 2 | +2 |
| `mapsEms` | - | 1 | +1 |
| `ml` | - | 14 | +14 |
| `reporting` | - | 1 | +1 |
| `savedObjects` | - | 3 | +3 |
| `share` | - | 1 | +1 |
| `snapshotRestore` | - | 1 | +1 |
| `triggersActionsUi` | - | 1 | +1 |
| `visualizations` | - | 11 | +11 |
| total |  |  | +363 |

#### Non-exported public API item count
| id | [before](https://github.com/elastic/kibana/commit/6c07eb238212b8d6b48579640e8374cef5046d36) | [after](https://github.com/elastic/kibana/commit/34d1650a6d0c8bd9c37c6150d0bd8f7441464991) | diff |
| --- | --- | --- | --- |
| `actions` | - | 6 | +6 |
| `advancedSettings` | - | 1 | +1 |
| `alerting` | - | 9 | +9 |
| `apm` | - | 8 | +8 |
| `bfetch` | - | 2 | +2 |
| `canvas` | - | 2 | +2 |
| `cases` | - | 1 | +1 |
| `charts` | - | 2 | +2 |
| `core` | - | 29 | +29 |
| `dashboard` | - | 9 | +9 |
| `data` | - | 75 | +75 |
| `dataEnhanced` | - | 2 | +2 |
| `devTools` | - | 2 | +2 |
| `discover` | - | 6 | +6 |
| `discoverEnhanced` | - | 2 | +2 |
| `embeddable` | - | 3 | +3 |
| `encryptedSavedObjects` | - | 4 | +4 |
| `esUiShared` | - | 4 | +4 |
| `eventLog` | - | 4 | +4 |
| `expressions` | - | 7 | +7 |
| `features` | - | 1 | +1 |
| `fleet` | - | 10 | +10 |
| `globalSearch` | - | 6 | +6 |
| `home` | - | 8 | +8 |
| `indexManagement` | - | 4 | +4 |
| `indexPatternFieldEditor` | - | 4 | +4 |
| `indexPatternManagement` | - | 4 | +4 |
| `infra` | - | 2 | +2 |
| `ingestPipelines` | - | 4 | +4 |
| `inspector` | - | 4 | +4 |
| `kibanaLegacy` | - | 1 | +1 |
| `kibanaReact` | - | 3 | +3 |
| `kibanaUtils` | - | 8 | +8 |
| `lens` | - | 12 | +12 |
| `licensing` | - | 10 | +10 |
| `lists` | - | 60 | +60 |
| `management` | - | 4 | +4 |
| `maps` | - | 16 | +16 |
| `ml` | - | 34 | +34 |
| `navigation` | - | 3 | +3 |
| `observability` | - | 8 | +8 |
| `presentationUtil` | - | 4 | +4 |
| `reporting` | - | 18 | +18 |
| `ruleRegistry` | - | 3 | +3 |
| `runtimeFields` | - | 2 | +2 |
| `savedObjects` | - | 5 | +5 |
| `savedObjectsManagement` | - | 1 | +1 |
| `security` | - | 10 | +10 |
| `securityOss` | - | 3 | +3 |
| `securitySolution` | - | 7 | +7 |
| `share` | - | 3 | +3 |
| `snapshotRestore` | - | 1 | +1 |
| `spaces` | - | 3 | +3 |
| `taskManager` | - | 3 | +3 |
| `telemetry` | - | 3 | +3 |
| `telemetryCollectionManager` | - | 4 | +4 |
| `timelines` | - | 1 | +1 |
| `triggersActionsUi` | - | 28 | +28 |
| `uiActions` | - | 9 | +9 |
| `uiActionsEnhanced` | - | 10 | +10 |
| `uptime` | - | 3 | +3 |
| `usageCollection` | - | 7 | +7 |
| `visTypeTimeseries` | - | 2 | +2 |
| `visualizations` | - | 6 | +6 |
| total |  |  | +520 |


</details>

### History
* :broken_heart: [Build #120056](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120056/) failed 84828e34f41a3fc0a8928733b95810a54dcc21c5
* :green_heart: [Build #119858](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119858/) succeeded bb8e2aafcd0d7e73742d6956521c9f693a4fc596
* :broken_heart: [Build #119838](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119838/) failed 62864b136641ba2cc346620bd6c75d8048a67cfb

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119838/","number":"119838","commit":"62864b136641ba2cc346620bd6c75d8048a67cfb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119858/","number":"119858","commit":"bb8e2aafcd0d7e73742d6956521c9f693a4fc596"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120056/","number":"120056","commit":"84828e34f41a3fc0a8928733b95810a54dcc21c5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120061/","number":"120061","commit":"34d1650a6d0c8bd9c37c6150d0bd8f7441464991"}]}
PIPELINE-->

jportner commented on src/core/server/http/security_response_headers_config.ts:
```diff
@@ -0,0 +1,72 @@
+    ],
+    { defaultValue: 'no-referrer-when-downgrade' }
+  ),
+  permissionsPolicy: schema.oneOf([schema.string(), schema.literal(null)], {
```
Sure, I was on the fence about this as well. I just wanted to do as much as possible to get ahead of scanning tools, but I think the only one that checks for this atm is securityheaders.com, and I think if you don't use this header it just caps your score at "A" (instead of "A+").

Perhaps we can leave the config property in, but change it to null/disabled by default?

legrego replied:
I'd still like to mark the property as experimental -- we don't know what the future holds for deprecating/removing features after the 8.0 release, so this would give us some more flexibility

legrego reviewed (commented):
One followup thought -- we should update [core telemetry](https://github.com/elastic/kibana/blob/d1040f01058914bd87620ddfe4da80d603f7423c/src/plugins/kibana_usage_collection/server/collectors/core/core_usage_collector.ts#L12) to indicate if these settings have been altered. It'd be great to track this over time in case we need data about future changes we want to make.

kobelb commented on src/core/server/http/security_response_headers_config.ts:
```diff
@@ -0,0 +1,72 @@
+export const securityResponseHeadersSchema = schema.object({
+  strictTransportSecurity: schema.oneOf([schema.string(), schema.literal(null)], {
+    // See: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security
+    defaultValue: 'max-age=31536000', // 1 year in seconds
```
How does `Strict-Transport-Security: max-age=31536000` affect Kibana when it's hosted on a base-path? For example, if Kibana is hosted at `https://foo.com/kibana/` will it apply to websites at `https://foo.com/bar`? If it's in effect for the entire domain, at a minimum, we won't want to set this in 7.13.

jportner replied:
http://gph.is/1lX4Z3B

I'll make an issue to change the defaults for `strictTransportSecurity` and `disableEmbedding` in the 8.0 release.

jportner commented:
Removed defaults for `strictTransportSecurity` and `permissionsPolicy`, and added usage data collection.

New securityheaders.com scan result:

![image](https://user-images.githubusercontent.com/5295965/114961133-6da6c280-9e36-11eb-84ba-cdff62ab06bd.png)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120238/)
* Commit: a0ecb67f8fbc0f4d2dd48ffab949a72ca74b0e73
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Doc API Changes](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120238/execution/node/416/log)

### History
* :green_heart: [Build #120061](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120061/) succeeded 34d1650a6d0c8bd9c37c6150d0bd8f7441464991
* :broken_heart: [Build #120056](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120056/) failed 84828e34f41a3fc0a8928733b95810a54dcc21c5
* :green_heart: [Build #119858](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119858/) succeeded bb8e2aafcd0d7e73742d6956521c9f693a4fc596
* :broken_heart: [Build #119838](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119838/) failed 62864b136641ba2cc346620bd6c75d8048a67cfb

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119838/","number":"119838","commit":"62864b136641ba2cc346620bd6c75d8048a67cfb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119858/","number":"119858","commit":"bb8e2aafcd0d7e73742d6956521c9f693a4fc596"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120056/","number":"120056","commit":"84828e34f41a3fc0a8928733b95810a54dcc21c5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120061/","number":"120061","commit":"34d1650a6d0c8bd9c37c6150d0bd8f7441464991"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120238/","number":"120238","commit":"a0ecb67f8fbc0f4d2dd48ffab949a72ca74b0e73"}]}
PIPELINE-->

jportner commented:
One API integration test is failing with the following message:
```
     │1)    apis
     │       Telemetry
     │         /api/telemetry/v2/clusters/_stats
     │           validate data types
     │             should pass the schema validation:
     │
     │      Error: The telemetry schemas in 'src/plugins/telemetry/schema/' are out-of-date, please update it as required: [stack_stats.kibana.plugins.core.config.http.securityResponseHeaders.strictTransportSecurity]: expected value of type [string] but got [null]
     │       at ObjectType.validate (packages/kbn-config-schema/src/types/type.ts:70:13)
     │       at assertTelemetryPayload (test/api_integration/apis/telemetry/utils/schema_to_config_schema.ts:137:34)
     │       at Context.apply (test/api_integration/apis/telemetry/telemetry_local.ts:58:11)
     │       at Object.apply (node_modules/@kbn/test/src/functional_test_runner/lib/mocha/wrap_function.js:73:30)
```

I set the type for this and some other security headers in the usage data config as `string | null`, and I set the ES field mapping to `"keyword"`. It seems that the existing schema validation does not support nullable values, though.

I can think of two obvious solutions:
1. Allow keyword and test values to be nullable:
   ```diff
   diff --git a/test/api_integration/apis/telemetry/utils/schema_to_config_schema.ts b/test/api_integration/apis/telemetry/utils/schema_to_config_schema.ts
   index b45930682e3..84409cfe1f7 100644
   --- a/test/api_integration/apis/telemetry/utils/schema_to_config_schema.ts
   +++ b/test/api_integration/apis/telemetry/utils/schema_to_config_schema.ts
   @@ -59,7 +59,7 @@ function valueSchemaToConfigSchema(value: TelemetrySchemaValue): Type<unknown> {
          case 'keyword':
          case 'text':
          case 'date':
   -        return schema.string();
   +        return schema.nullable(schema.string());
          case 'byte':
          case 'double':
          case 'float':
   ```
2. Convert `null` values to `"NULL"` strings in the usage collector, and change the `string | null` type to `string`

@elastic/kibana-telemetry what do you think?

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120243/)
* Commit: d06cd8c04d4de3701f0d157d06ae981f573c73df
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [API Integration Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120243/execution/node/420/log)

### History
* :broken_heart: [Build #120238](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120238/) failed a0ecb67f8fbc0f4d2dd48ffab949a72ca74b0e73
* :green_heart: [Build #120061](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120061/) succeeded 34d1650a6d0c8bd9c37c6150d0bd8f7441464991
* :broken_heart: [Build #120056](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120056/) failed 84828e34f41a3fc0a8928733b95810a54dcc21c5
* :green_heart: [Build #119858](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119858/) succeeded bb8e2aafcd0d7e73742d6956521c9f693a4fc596
* :broken_heart: [Build #119838](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119838/) failed 62864b136641ba2cc346620bd6c75d8048a67cfb

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119838/","number":"119838","commit":"62864b136641ba2cc346620bd6c75d8048a67cfb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119858/","number":"119858","commit":"bb8e2aafcd0d7e73742d6956521c9f693a4fc596"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120056/","number":"120056","commit":"84828e34f41a3fc0a8928733b95810a54dcc21c5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120061/","number":"120061","commit":"34d1650a6d0c8bd9c37c6150d0bd8f7441464991"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120238/","number":"120238","commit":"a0ecb67f8fbc0f4d2dd48ffab949a72ca74b0e73"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120243/","number":"120243","commit":"d06cd8c04d4de3701f0d157d06ae981f573c73df"}]}
PIPELINE-->

pgayvallet reviewed (approved):
Technically, LGTM

jportner commented on src/core/server/csp/csp_config.ts:
```diff
@@ -6,9 +6,12 @@
+import { config, FRAME_ANCESTORS_RULE } from './config';
 
-const DEFAULT_CONFIG = Object.freeze(config.schema.validate({}));
+const DEFAULT_CONFIG = Object.freeze({
+  ...config.schema.validate({}),
+  disableEmbedding: false,
+});
```
Yes, I didn't want to have two conflicting config properties -- I didn't consider this idea. I'll change it and add a comment!

For future reference, when I first tried this change I got a type error (`boolean` is not equivalent to `false`), but luckily `schema.literal` is a generic so I got around it with this:

```schema.literal<boolean>(false)```

😄

jportner commented:
> I can think of two obvious solutions:
> 
> 1. Allow keyword and test values to be nullable in the API integration tests:
>    ```diff
>     ...
>    ```
> 2. Convert `null` values to `"NULL"` strings in the usage collector, and change the `string | null` type to `string`
> 
> @elastic/kibana-telemetry what do you think?

I spoke to @Bamieh and @afharo offline and this point was raised:

> my only concern with allowing `null` to types is that it'll bring more unexpected scenarios that it'll do good. it is also mean that we'd need to treat every `null` we get in the ES indexing since ES does not index null and it cannot be searched.

So we agreed that option (2) above is the best path forward. I updated this accordingly in a501b8c.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120589/)
* Commit: a501b8c61abfb1571148dc3a0af07ad6dd0883af
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120589/execution/node/414/log)

### History
* :broken_heart: [Build #120243](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120243/) failed d06cd8c04d4de3701f0d157d06ae981f573c73df
* :broken_heart: [Build #120238](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120238/) failed a0ecb67f8fbc0f4d2dd48ffab949a72ca74b0e73
* :green_heart: [Build #120061](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120061/) succeeded 34d1650a6d0c8bd9c37c6150d0bd8f7441464991
* :broken_heart: [Build #120056](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120056/) failed 84828e34f41a3fc0a8928733b95810a54dcc21c5
* :green_heart: [Build #119858](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119858/) succeeded bb8e2aafcd0d7e73742d6956521c9f693a4fc596

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/119858/","number":"119858","commit":"bb8e2aafcd0d7e73742d6956521c9f693a4fc596"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120056/","number":"120056","commit":"84828e34f41a3fc0a8928733b95810a54dcc21c5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120061/","number":"120061","commit":"34d1650a6d0c8bd9c37c6150d0bd8f7441464991"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120238/","number":"120238","commit":"a0ecb67f8fbc0f4d2dd48ffab949a72ca74b0e73"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120243/","number":"120243","commit":"d06cd8c04d4de3701f0d157d06ae981f573c73df"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120589/","number":"120589","commit":"a501b8c61abfb1571148dc3a0af07ad6dd0883af"}]}
PIPELINE-->

afharo commented on src/plugins/telemetry/schema/oss_plugins.json:
```diff
@@ -6742,6 +6742,40 @@
+                    "strictTransportSecurity": {
+                      "type": "keyword",
+                      "_meta": {
+                        "description": "The strictTransportSecurity response header, if it is configured."
```
> if it is configured

I don't think this is true. We are always reporting it. Either the configured value or `"NULL"`. Maybe I misread the code though. Can you confirm @jportner?

jportner replied:
Oh, yes sorry about that, I forgot to update the descriptions when I added the `null` coalescing. I'll update those now.

afharo reviewed (commented):
Just one note around the descriptions :)

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120603/)
* Commit: fd7874580a3e9410b1f9246b5d5a44f763db7144
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-97158/fd7874580a3e9410b1f9246b5d5a44f763db7144)
* [Documentation Changes](https://kibana_97158.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/1b02152e185caf3cfb8d997558650a0d388be4a2) | [after](https://github.com/elastic/kibana/commit/fd7874580a3e9410b1f9246b5d5a44f763db7144) | diff |
| --- | --- | --- | --- |
| `core` | 2172 | 2174 | +2 |

#### API count missing comments
| id | [before](https://github.com/elastic/kibana/commit/1b02152e185caf3cfb8d997558650a0d388be4a2) | [after](https://github.com/elastic/kibana/commit/fd7874580a3e9410b1f9246b5d5a44f763db7144) | diff |
| --- | --- | --- | --- |
| `core` | 991 | 992 | +1 |


</details>

### History
* :broken_heart: [Build #120589](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120589/) failed a501b8c61abfb1571148dc3a0af07ad6dd0883af
* :broken_heart: [Build #120243](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120243/) failed d06cd8c04d4de3701f0d157d06ae981f573c73df
* :broken_heart: [Build #120238](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120238/) failed a0ecb67f8fbc0f4d2dd48ffab949a72ca74b0e73
* :green_heart: [Build #120061](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120061/) succeeded 34d1650a6d0c8bd9c37c6150d0bd8f7441464991
* :broken_heart: [Build #120056](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120056/) failed 84828e34f41a3fc0a8928733b95810a54dcc21c5

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120056/","number":"120056","commit":"84828e34f41a3fc0a8928733b95810a54dcc21c5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120061/","number":"120061","commit":"34d1650a6d0c8bd9c37c6150d0bd8f7441464991"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120238/","number":"120238","commit":"a0ecb67f8fbc0f4d2dd48ffab949a72ca74b0e73"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120243/","number":"120243","commit":"d06cd8c04d4de3701f0d157d06ae981f573c73df"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120589/","number":"120589","commit":"a501b8c61abfb1571148dc3a0af07ad6dd0883af"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/120603/","number":"120603","commit":"fd7874580a3e9410b1f9246b5d5a44f763db7144"}]}
PIPELINE-->

kibanamachine commented:
## 💔 Backport failed

| Status | Branch | Result |
|:------:|:------:|:------:|
| ❌ |  7.x  | Commit could not be cherrypicked due to conflicts |

To backport manually run:
`node scripts/backport --pr 97158`

elasticmachine commented:
Pinging @elastic/kibana-security (Team:Security)
